mmocr/models/textrecog/backbones/resnet.py

```diff
@@ -0,0 +1,232 @@
+
+        Examples:
+            >>> plugins=[
+            ...     dict(cfg=dict(type="Maxpooling", arg=(2,2)),
+            ...          stages=(True, True, False, False),
+            ...          position='before_stage'),
+            ...     dict(cfg=dict(type="Maxpooling", arg=(2,1)),
+            ...          stages=(False, False, True, Flase),
+            ...          position='before_stage'),
+            ...     dict(cfg=dict(type="Conv", args=(3,1,1)),
+            ...          stages=(True, True, True, True),
+            ...          position=('after_stage')
+            ... ]
+
+        Suppose ``stage_idx=1``, the structure of stage would be:
+
+        .. code-block:: none
+
+            Maxpooling-> Basicblocks * blocks[stage_idx - 1] -> Conv
+
+        Args:
+            plugins (list[dict]): List of plugins cfg to build. The postfix is
+                required if multiple same type plugins are inserted.
+            stage_idx (int): Index of stage to build
+
+        Returns:
+            list[dict]: Plugins for current stage
+        """
+        in_channels = self.arch_channels[stage_idx]
+        self.plugin_ahead_names.append([])
+        self.plugin_after_names.append([])
+        for plugin in plugins:
+            plugin = plugin.copy()
+            stages = plugin.pop('stages', None)
+            position = plugin.pop('position', None)
+            assert stages is None or len(stages) == self.num_stages
+            if stages[stage_idx]:
+                if position == 'before_stage':
+                    name, layer = build_plugin_layer(
+                        plugin['cfg'],
+                        f'_before_stage_{stage_idx+1}',
+                        in_channels=in_channels,
+                        out_channels=in_channels)
+                    self.plugin_ahead_names[stage_idx].append(name)
+                    self.add_module(name, layer)
+                elif position == 'after_stage':
+                    name, layer = build_plugin_layer(
+                        plugin['cfg'],
+                        f'_after_stage_{stage_idx+1}',
+                        in_channels=in_channels,
+                        out_channels=in_channels)
+                    self.plugin_after_names[stage_idx].append(name)
+                    self.add_module(name, layer)
+                else:
+                    raise ValueError('uncorrect plugin position')
+
+    def forward_plugin(self, x, plugin_name):
+        out = x
+        for name in plugin_name:
+            out = getattr(self, name)(x)
+        return out
+
+    def forward(self, x):
+        """Args:p x (Tensor): Image tensor of shae :math:`(N, 3, H, W)`.
+
+        Returns:
+            Tensor or list[Tensor]: Feature tensor. Its shape depends on
+            ResNetABI's config. It can be a list of feature outputs at specific
+            layers if ``out_indices`` is specified.
+
+            output (Tensor): shape :math: `(N, 512, H/4, W/4)`
```

Comment:
We usually write one paragraph only in this section.

Reply:
I should remove the shape description of output tensor,